src/OptFrame/RandGen.hpp

```diff
@@ -111,7 +112,11 @@ class RandGen : public Component
    // initialize random number generation
    virtual void initialize()
    {
-      _gen = std::mt19937(this->seed);
+      unsigned _seed = this->seed;
+      std::array<decltype(_seed), _gen.state_size> seed_data;
+      std::generate(seed_data.begin(), seed_data.end(), [&_seed](){ return _seed++; });
+      std::seed_seq seed_sequence(seed_data.begin(), seed_data.end());
+      _gen.seed(seed_sequence);
```

Comment:
Do you have a reference link @AndersonZM ? So it's better to inject several incremental seeds than standard one?

Reply:
Check [Mars answer](https://stats.stackexchange.com/questions/436733/is-there-such-a-thing-as-a-good-bad-seed-in-pseudo-random-number-generation) in StackExchange. He made a short summary of why we should adequately initialize the Mersenne Twister algorithm while linking several studies on the matter. 

He explains: "[...] one of the issues with Mersenne Twisters is that if its internal 624x32-bit state has many zero bits, it takes many iterations to get out of that pattern."

When you look into the implementation of std::, one can see that the constructor generates at least one random number for each seed given. The static variable (engine size) from std::mt19937 gives the standard engine size, in this case, 624. By providing that many seeds to the engine, the quality of the following iterations is significantly higher and, in most cases, independent of the seed size.